src/RSocketRequester.cpp

```diff
@@ -35,6 +35,7 @@ RSocketRequester::RSocketRequester(
 
 RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
+  stateMachine_->close(folly::exception_wrapper(), StreamCompletionSignal::CONNECTION_END);
```

Comment:
yep, this is a correct fix. We were not calling close.

> I saw that in the close() method of the RSocketStateMachine, we call lots of callback functions. Any of these functions might fail and if might cause early termination of the function. But at the start of the function we call

The current solution for this is that the callbacks are defined with noexcept keyword which means if there is an exception bubbling back to the caller, it will std::terminate the whole program. Maybe not ideal solution, but it eliminates the problem of early termination..